File: src/main/java/com/clearcont/clearcontapp/service/CustomerContabilService.java
package com.clearcont.clearcontapp.service;

import com.clearcont.clearcontapp.model.CustomerContabil;
import com.clearcont.clearcontapp.repository.CustomerContabilRepository;
import jakarta.transaction.Transactional;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@Transactional
public class CustomerContabilService {
    private final CustomerContabilRepository repository;

    public CustomerContabilService(CustomerContabilRepository repository) {
        this.repository = repository;
    }

    @Transactional
    public void save(CustomerContabil customer) {
        repository.save(customer);
    }

    public List<CustomerContabil> findByBalanceteID(Long id) {
        return repository.findAllByComposicaoLancamentosContabeis_Balancete_Id(id);
    }

    public CustomerContabil update(CustomerContabil customerContabil) {
        return repository.saveAndFlush(customerContabil);
    }
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/service/CustomerContabilService.kt
package br.com.clearcont.clearcontwebapp.service

import br.com.clearcont.clearcontwebapp.models.CustomerContabil
import br.com.clearcont.clearcontwebapp.models.Responsavel
import br.com.clearcont.clearcontwebapp.models.enums.TipoConta
import br.com.clearcont.clearcontwebapp.repository.CustomerContabilRepository
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
@Transactional
class CustomerContabilService(
    private val repository: CustomerContabilRepository,
    private val balanceteService: BalanceteService
) {
    @Transactional
    fun save(customerContabil: CustomerContabil, balanceteId: Long, responsavel: Responsavel, tipoConta: TipoConta) {
        customerContabil.composicaoLancamentosContabeis.balancete = balanceteService.getById(balanceteId)
        customerContabil.composicaoLancamentosContabeis.balancete!!.tipo =  tipoConta
        customerContabil.composicaoLancamentosContabeis.responsavel = responsavel
        repository.save(customerContabil)
    }

    fun findByBalanceteID(id: Long): List<CustomerContabil> {
        return repository.findCustomerContabilByBalanceteId(id)
    }

    fun update(customerContabil: CustomerContabil): CustomerContabil {
        return repository.saveAndFlush(customerContabil)
    }
}
